networkit/cpp/io/FastSNAPGraphReader.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * FastSNAPGraphReader.h
+ *
+ *  Created on: 04.05.2018
+ *      Author: Alexander van der Grinten
+ */
+#ifndef FASTSNAPGRAPHREADER_H_
+#define FASTSNAPGRAPHREADER_H_
+
+#include <unordered_map>
+
+#include "../graph/Graph.h"
+#include "GraphReader.h"
+
+namespace NetworKit {
+
+/**
+ * @ingroup io
+ */
+class FastSNAPGraphReader : public NetworKit::GraphReader {
+private:
+	std::unordered_map<node, node> nodeIdMap;
+	bool directed;
+	count nodeCount;
+	bool remapNodes;
+
+public:
+	/**
+	 * Default constructor for the FastSNAPGraphReader.
+	 * NOTE: Keep in mind that many SNAP graphs do not have consecutive node ids.
+	 * Passing a nodeCount might result in a higher memory use for the output graph
+	 * but reduces runtime. Leaving this parameter on its default value, the reader
+	 * is forced to remap the nodes and thereby decreases the memory use on cost of
+	 * slighty higher runtime.
+	 *
+	 * @param[in]	directed	reads in the graph as directed, if set to true
+	 * @param[in]	nodeCount	nodeCount is used to preallocated memory for the number of nodes
+	 * @param[in]	remapNodes	indicates whether nodes should be remapped to other node ids in order to create consecutive node ids
+	 */
+	FastSNAPGraphReader(const bool& directed = false, const count& nodeCount = 0, const bool& remapNodes = true);
```

Comment:
Passing a `bool` as const reference is not really useful as here you replace a 1 byte value by an 8 byte pointer. I would suggest to just pass them by value.

Reply:
True, I fixed it in https://github.com/kit-parco/networkit/pull/168/commits/f944ddb670fdac2161c583d90acf6a2e0414604d